### PR TITLE
Add for update clause

### PR DESCRIPTION
**Pull request by MooreGuy**
Allows the use of the FOR UPDATE clause in select statements. Doesn't
have anything special to it, just needs to be at the end of the query
and only available to select statements.

QA: You can make some queries with for updates!

### CODE REVIEW

**Comment by djmetzle** [comment]
CR :+1:

**Comment by djmetzle** [comment]
CR :mag: again

**Comment by danielbeardsley** [comment]
CR 👍

File: QueryGeneratorTest.php [hunk 1]
```diff
@@ -208,6 +208,21 @@ public function testModifiers() {
...
+
+      $expectedQuery = <<<EOT
+SELECT a
+FROM b
+FOR UPDATE
+EOT;
+
+      $this->assertQuery($qGen, $expectedQuery, []);
+   }
+
```
Comment (MooreGuy) [inline QueryGeneratorTest.php]: Thanks for adding this!

**Comment by djmetzle** [comment]
Another one. CR :mag:

**Comment by MooreGuy** [comment]
Is it fine if I call QA on this as being used in the pull I am currently working on?

**Comment by MooreGuy** [comment]
It works as intended there.

**Comment by danielbeardsley** [comment]
> Is it fine if I call QA on this as being used in the pull I am currently working on?

Sounds fine by me. Passes tests.

**Comment by MooreGuy** [comment]
QA :+1: from using it in my own code

**Pull request merged**